### DISCUSSION TITLE
画一段很长的path越来越卡，请问怎么优化

### DISCUSSION THREAD

**Discussion by ghbhaha** [Tencent/tgfx#112] [Q&A] [answered]
安卓端，Path在触摸move时不断的lineto 绘制会越来越卡，请问可以优化么

**Comment by ghbhaha** [comment]
https://github.com/Tencent/tgfx/assets/5380881/01d14d76-14e6-4d5f-96a0-fb2c403cd1ee

**Reply by domchen** [reply, marked as answer]
你是不是一直在往同一个Path对象上不断累加数据然后每次清屏重画。这样操作无论是什么方案最后都会触发性能临界值。你的还是靠无数的直线在模拟曲线，数据量巨大。这种场景要用起来缓存，不要清屏，每次只生成新增的 Path画上去就行了。要缓存也用Suface.makeImageSnapshot()生成画好的图片缓存，不应该去缓存无限大的矢量数据。

**Reply by ghbhaha** [reply]
嗯嗯 谢谢 是跟这个相关吧AA_TESSELLATOR_MAX_VERB_COUNT

**Reply by domchen** [reply]
你这是个业务逻辑问题，跟内部的常量没关系。这个常量是判断用软件栅格化还是拆三角形的临界值。就算全部拆三角形，你积累一个无限大的Path怎么样都是要卡顿的。重点是之前绘制过的内容不应该积累起来再绘制一次。每次就只画增量才是对的。

**Reply by ghbhaha** [reply]
👌🏻
